packages/contracts-bedrock/STYLE_GUIDE.md

```diff
@@ -104,9 +104,15 @@ Unless explicitly discussed otherwise, you MUST include the following basic upgr
 pattern for each new implementation contract:
 
 1. Extend OpenZeppelin's `Initializable` base contract.
-2. Include a `uint8 public constant VERSION = X` at the TOP of your contract.
-3. Include a function `initialize` with the modifier `reinitializer(VERSION)`.
-4. In the `constructor`, set any `immutable` variables and call the `initialize` function for setting mutables.
+2. Include a function `initialize` with the modifier `initializer()`.
+3. In the `constructor`:
+    1. Call `_disableInitializers()` to ensure the implementation contract cannot be initialized.
+    2. Set any immutables.
+
+Because `reinitializer(uint64 version)` is not used, the process for upgrading the implementation is:
```

Comment:
For the avoidance of doubt:

```suggestion
Because `reinitializer(uint64 version)` is not used, the process for upgrading the implementation is to atomically:
```